File: Javascript/QuickSort.js
function quickSort (array) {
    if (array.length > 1) {
        let pivot = array.length - 1
        let smallArray = []
        let bigArray = []
        for (let item = 0; item < array.length - 1; item += 1) {
            if (array[item] <= array[pivot]) smallArray.push(array[item])
            else bigArray.push(array[item])
        }

        return quickSort(smallArray).concat([array[pivot]], quickSort(bigArray))
    }
    else return array
}

File: Python/QuickSort.py
def quickSort(array):
    if len(array) > 1:
        pivot = len(array) - 1
        smallArray = []
        bigArray = []
        for item in range(len(array) - 1):
            if array[item] <= array[pivot]:
                smallArray.append(array[item])
            else:
                bigArray.append(array[item])
        return quickSort(smallArray) + [array[pivot]] + quickSort(bigArray)
    else:
        return array
